viewing_party/party.py

```diff
@@ -1,23 +1,149 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == title:
+            movie_watched = user_data["watchlist"][i]
+            del user_data["watchlist"][i]
+            user_data["watched"].append(movie_watched)  
+    return user_data
+
+    # .haskey() does NOT work in python3 --> use key in list of dictionary
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    sum_rating = 0
+    count = 0
+    for i in range(len(user_data["watched"])):
+        sum_rating += user_data["watched"][i]["rating"]
+        count += 1
+    if count == 0:
+        return 0
+    else:
+        return sum_rating/count
+
+def get_most_watched_genre(user_data):
+    # make a list of all the genres user watched
+    genre_list = []
+    for i in range(len(user_data["watched"])):
+        genre_list.append(user_data["watched"][i]["genre"])
+    # if user did not watch many movies or if genre key has no value, return None
+    if len(genre_list) == 0:
+        return None
+    # use a for loop and a counter to find the most frequent genre
+    current_max_frequency = 0
+    max_genre = ""
+    for genre in genre_list:
+        genre_frequency = genre_list.count(genre)
+        if genre_frequency > current_max_frequency:
+            current_max_frequency = genre_frequency
+            max_genre = genre
+    return max_genre
+    
+    # I can also add if statements to account for multiple genres.
+    # alternative method: return max(set(genre_list), key=genre_list.count)
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    # make friends_movie_list 
+    friends_movie_list = []
+    for dictionary in user_data["friends"]:
+        for movie in dictionary["watched"]:
+            if movie not in friends_movie_list:
+                friends_movie_list.append(movie)
+    # alternative way: friends_movie_lists = [d["watched"] for d in user_data["friends"]] 

```

Comment:
I like how you considered an alternative here! I think in the alternative there might be a bug though, I think you would need to flatten this as it would end up a list of lists.